MapboxCoreNavigationTests/MapboxCoreNavigationTests.swift

```diff
@@ -224,6 +224,85 @@ class MapboxCoreNavigationTests: XCTestCase {
+            let routeProgress = notification.userInfo![RouteControllerNotificationUserInfoKey.routeProgressKey] as! RouteProgress
+            let legIndex = routeProgress.legIndex
+            let stepIndex = routeProgress.currentLegProgress.stepIndex
+            let spokenInstructionIndex = routeProgress.currentLegProgress.currentStepProgress.spokenInstructionIndex
+            let visualInstructionIndex = routeProgress.currentLegProgress.currentStepProgress.visualInstructionIndex
+            
+            let point = InstructionPoint(type: .spoken, legIndex: legIndex, stepIndex: stepIndex, spokenInstructionIndex: spokenInstructionIndex, visualInstructionIndex: visualInstructionIndex)
+            points.append(point)
+            
+            return true
+        }
+        
+        expectation(forNotification: .routeControllerDidPassVisualInstructionPoint, object: nil) { (notification) -> Bool in
+            let routeProgress = notification.userInfo![RouteControllerNotificationUserInfoKey.routeProgressKey] as! RouteProgress
+            let legIndex = routeProgress.legIndex
+            let stepIndex = routeProgress.currentLegProgress.stepIndex
+            let spokenInstructionIndex = routeProgress.currentLegProgress.currentStepProgress.spokenInstructionIndex
+            let visualInstructionIndex = routeProgress.currentLegProgress.currentStepProgress.visualInstructionIndex
+            
+            let point = InstructionPoint(type: .visual, legIndex: legIndex, stepIndex: stepIndex, spokenInstructionIndex: spokenInstructionIndex, visualInstructionIndex: visualInstructionIndex)
+            points.append(point)
+            
+            return true
+        }
+        
+        for location in trace {
+            navigation.router!.locationManager!(navigation.locationManager, didUpdateLocations: [location])
+        }
+        
+        waitForExpectations(timeout: 2) { (error) in
+            XCTAssertNil(error)
+        }
+        
+        XCTAssertEqual(points[0].legIndex, 0)
+        XCTAssertEqual(points[0].stepIndex, 0)
+        XCTAssertEqual(points[0].visualInstructionIndex, 0)
+        XCTAssertEqual(points[0].spokenInstructionIndex, 0)
+        XCTAssertEqual(points[0].type, .spoken)
+        
+        XCTAssertEqual(points[1].legIndex, 0)
+        XCTAssertEqual(points[1].stepIndex, 0)
+        XCTAssertEqual(points[1].visualInstructionIndex, 0)
+        XCTAssertEqual(points[1].spokenInstructionIndex, 0)
+        XCTAssertEqual(points[1].type, .visual)
+        
+        XCTAssertEqual(points[2].legIndex, 0)
+        XCTAssertEqual(points[2].stepIndex, 0)
+        XCTAssertEqual(points[2].visualInstructionIndex, 0)
+        XCTAssertEqual(points[2].spokenInstructionIndex, 1)
+        XCTAssertEqual(points[2].type, .spoken)
+        
+        XCTAssertEqual(points[3].legIndex, 0)
+        XCTAssertEqual(points[3].stepIndex, 1)
```

Comment:
This assertion fails because stepIndex is 0, indicating that we are going backward in time, spokenInstructionIndex went from 1 to 0 when stepIndex remained on 0.